### ISSUE TITLE
Based on an E-Mail regarding multiple Azure Config Files

### ISSUE THREAD

**Issue by MarkWarneke** [MarkWarneke/Az.Cli#10]
**Describe the bug**

> I work with multiple tenants / accounts and to make my life easier I use the following configuration:
> 
> BASH_FUNC_az_uar%%=() {  AZURE_CONFIG_DIR=/home/user/.azure_uat az "$@"
> BASH_FUNC_az_prd%%=() {  AZURE_CONFIG_DIR=/home/user/.azure_prd az "$@"
> BASH_FUNC_az_sit%%=() {  AZURE_CONFIG_DIR=/home/user/.azure_sit az "$@"
> 
> So I can run the commands without having to log in to each tenant!
> 
> az_uat ad app show --id XXXXXXXXXX
> az_prd ad app show --id XXXXXXXXXX
> az_sit ad app show --id XXXXXXXXXX
> 
> Unfortunately in a script when trying to run the commands in different tenants I got authentication errors. I tried to adjust the AZURE_CONFIG_DIR variable before loading the module, I also tried using importlib and reloading the module but nothing worked!
> 
> As a last solution, I started using the subprocess module and ran each command in a Python interpreter and got the value using a Queue. That way it worked!
> 
> But as a curiosity, I would like to know if there is a way to adjust AZURE_CONFIG_DIR programmatically from code!
> 

**To Reproduce**

Create multiple az context

```
az login

mv ~/.azure/* ~/.azure-mw
az login --service-principal -u $id -p $p -t $t

# Validate this works
AZURE_CONFIG_DIR=.azure az group list 
AZURE_CONFIG_DIR=.azure-mw az group list 
``` 

**Expected behavior**

When changing the environment variable `AZURE_CONFIG_DIR` the command should not fail and use the at the point in time configured Azure config.

**Comment by MarkWarneke** [comment]
From what I understand is that you want to work on different Azure tenants using python?
Hence you created multiple configuration files that are located in ~/.azure_* which you want to use to invoke a python script.
Now you want to change the specific config file using Python code too?

I just validated it and I think I have a bug in my code. Similar to #8 .
To fix it now you need to tinker a bit with the imports, which is not really pythonic.
I think I already found a fix for this. I will update the package once I completed my tests. 

The main idea going forward would be that you have to [set the environment variable](https://docs.microsoft.com/en-us/cli/azure/use-cli-effectively?view=azure-cli-latest#cli-environment-variables) like this:

```python
os.environ['AZURE_CONFIG_DIR'] = '/Users/mark/.azure_mw'
```

**Comment by MarkWarneke** [comment]
I am thinking about adding a smaller helper for this, what do you think about the following method?

```python
from az.cli import az, set_az_config
set_az_config('/Users/mark/.azure_mw')
```

**Comment by MarkWarneke** [comment]
After testing version 0.5 I can confirm the issue has been resolved by introducing change #9 .

```bash
python3 -m venv env  
pip install 'az.cli==0.5' 
python3
```

```python
>>> from az.cli import az
>>> import os

>>> exit_code, result_dict, logs = az("group list")
>>> print (result_dict)
[{'id': '/subscriptions/...', 'location': 'westeurope',  'name': 'test1']

>>> os.environ['AZURE_CONFIG_DIR'] = '/Users/mark/.azure_mw'
>>> exit_code, result_dict, logs = az("group list")
>>> print (result_dict)
[{'id': '/subscriptions/...', 'location': 'westeurope', 'name': 'test2']
```

The outputs confirms that one execution is returning different values than the other based on `AZURE_CONFIG_DIR`.
